File: csharp/squeaky-clean/SqueakyClean.cs
using System;
using System.Text;

public static class Identifier
{
    public static string Clean(string identifier)
    {
        StringBuilder cleanString = new StringBuilder();
        int i = 0;
        char[] chars = identifier.ToCharArray();
        while (i < chars.Length)
        {
            char c = chars[i++];
            if (isGreekLowercase(c))
            {
                // omit
            }
            else if (Char.IsLetter(c))
            {
                cleanString.Append(c);
            }
            else if (Char.IsWhiteSpace(c))
            {
                cleanString.Append("_");
            }
            else if (Char.IsControl(c))
            {
                cleanString.Append("CTRL");
            }
            else if (c == '-')
            {
                cleanString.Append(Char.ToUpper(chars[i++]));
            }
        }

        return cleanString.ToString();
    }

    private static bool isGreekLowercase(char c) => c >= '\u03B1' && c <= '\u03C9';
}

File: java/squeaky-clean/src/main/java/SqueakyClean.java
import java.text.CharacterIterator;
import java.text.StringCharacterIterator;

class SqueakyClean {
    static String clean(String identifier) {
        StringBuilder cleanIdentifier = new StringBuilder();

        CharacterIterator it = new StringCharacterIterator(identifier);
        Character current = it.current();
        while (current != CharacterIterator.DONE) {
            if (isCamelCase(it)) {
                cleanIdentifier.append(Character.toUpperCase(it.current()));
            }
            if (Character.isLetter(current) && !isGreekLowerLetter(current)) {
                cleanIdentifier.append(current);
            }
            if (Character.isSpaceChar(current)) {
                cleanIdentifier.append("_");
            }
            if (Character.isISOControl(current)) {
                cleanIdentifier.append("CTRL");
            }
            current = it.next();
        }

        return cleanIdentifier.toString();
    }

    private static boolean isCamelCase(CharacterIterator it) {
        return it.current() == '-' && Character.isLetter(it.next());
    }

    private static boolean isGreekLowerLetter(Character letter) {
        return letter >= 'α' && letter <= 'ω';
    }
}
